File: exams/1617/translate/layout/solution/layout/box.cpp
#include "box.h"


Box::Box() : Box(0,0) { }

Box::Box(double width, double height)
	: width(width), height(height) { }

std::ostream& operator <<(std::ostream& out, const Box& box)
{
	return out << "Box(" << box.width << "x" << box.height << ")";
}

File: exams/1617/translate/layout/student/java/Box.java
public class Box
{
    public double width, height;
    
    public Box()
    {
        this(0, 0);
    }

    public Box(double width, double height)
    {
        this.width = width;
        this.height = height;
    }

    public Box(Box box)
    {
        this(box.width, box.height);
    }

    // Te implementeren als << op std::ostream
    @Override
    public String toString()
    {
        return "Box(" + width + "x" + height + ")";
    }
}
